### PR TITLE
Remove ND Derivation

### PR DESCRIPTION
**Pull request by jhand2**
DICE deterministic key derivation is now considered sufficient for all DPE use-cases. Remove the ND Derviation feature.

### CODE REVIEW

File: dpe/src/dpe_instance.rs [hunk 1]
```diff
@@ -32,12 +32,13 @@ pub struct DpeInstance<'a, C: Crypto, P: Platform> {
...
+    phantom_crypto: PhantomData<C>,
 }
 
 impl<C: Crypto, P: Platform> DpeInstance<'_, C, P> {
     const MAX_NEW_HANDLE_ATTEMPTS: usize = 8;
 
-    pub(crate) fn new_context_handles() -> [Context<C>; MAX_HANDLES] {
+    pub(crate) fn new_context_handles() -> [Context; MAX_HANDLES] {
         core::array::from_fn(|_| Context::new())
     }
```
Comment (zhalvorsen) [inline dpe/src/dpe_instance.rs:43]: nit: we could change this back to the `const` implementation we used to have

```Rust
const CONTEXT_INITIALIZER: Context = Context::new();
        let mut dpe = DpeInstance {
            contexts: [CONTEXT_INITIALIZER; MAX_HANDLES],
```

File: dpe/src/support.rs [hunk 2]
```diff
@@ -57,17 +55,14 @@ impl Support {
     fn get_is_symmetric_flag(&self) -> u32 {
         u32::from(self.is_symmetric) << 24
     }
```
Comment (zhalvorsen) [inline dpe/src/support.rs:57]: Do we need to reserve this bit as well?

**Review by zhalvorsen — APPROVED** [review]

File: dpe/src/dpe_instance.rs (hunk 1 above)
Comment (jhand2) [reply dpe/src/dpe_instance.rs:43]: Done.

File: dpe/src/support.rs (hunk 2 above)
Comment (zhalvorsen) [reply dpe/src/support.rs:57]: Nevermind, I misunderstood something

File: dpe/src/dpe_instance.rs [hunk 3]
```diff
@@ -32,12 +32,13 @@ pub struct DpeInstance<'a, C: Crypto, P: Platform> {
...
     // is just to make the Crypto and Platform traits work.
     phantom_platform: PhantomData<P>,
+    phantom_crypto: PhantomData<C>,
 }
 
 impl<C: Crypto, P: Platform> DpeInstance<'_, C, P> {
     const MAX_NEW_HANDLE_ATTEMPTS: usize = 8;
 
-    pub(crate) fn new_context_handles() -> [Context<C>; MAX_HANDLES] {
+    pub(crate) fn new_context_handles() -> [Context; MAX_HANDLES] {
```
[the remark this answers is not in the document]
Comment (jhand2) [reply dpe/src/dpe_instance.rs]: Done

**Review by zhalvorsen — APPROVED** [review]

**Pull request merged**